runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutor.java

```diff
@@ -400,13 +409,19 @@ void updateProgress() {
      * @param monitoringInfos Usually received from FnApi.
      */
     private void updateMetrics(List<MonitoringInfo> monitoringInfos) {
+      List<MonitoringInfo> monitoringInfosCopy = new ArrayList<>(monitoringInfos);
+
+      List<MonitoringInfo> misToFilter =
+          bundleProcessOperation.findIOPCollectionMonitoringInfos(monitoringInfos);
```

Comment:
Is there a reason why we need to collect counters for these ones at all? Our UI doesn't display the grpc steps, they are an implementation detail.

The alternative design I was thinking of here was to try and transform each monitoring info, and drop the ones that do not have a step in the original graph. Though, maybe there is no way to detect this? So they could be dropped int the transformer, as there is no need to send them to DFE.

If I understand correctly, you are just avoiding a double count now? Though I think we don't need to report these at all.

Reply:
Synced offline. I'll try to elaborate more on this.

When we modify graph for cross-boundary grpc operations, we give two different PCollections same metadata. As a result we report metrics correctly from SDK harness. Both of these PCollections generate metrics that show as ElementCount metric on UI.

At this point, the shortest way to fix the issue is to utilize deduping on runner.